Add the Transact-SQL (T-SQL) language

Greg-Smulko opened the pull request:
This PR adds the support for the Transact-SQL (T-SQL) dialect of the SQL language.

Before this PR, the T-SQL was highlighted like this:
<kbd>
![image](https://cloud.githubusercontent.com/assets/2973219/26199117/c36920c8-3bc0-11e7-983b-af56ebcbd4db.png)
</kbd>


There is a lot of improvements, for instance the `*/` comment closing character is now part of the comment, `XACT_ABORT` and `EXEC` are now keywords, `counted` is no longer a keyword, identifiers can be escaped with square brackets, the `N` prefix is now part of strings, identifiers can be escaped with double quotes.
So it improves to looking like this:
<kbd>
![image](https://cloud.githubusercontent.com/assets/2973219/26200233/15ad2358-3bc5-11e7-95af-5646772fab95.png)
</kbd>


Additionally, there is a new `ssms` styling created, which changes the colours to be identical to SSMS.
So it improves further to looking like this:
<kbd>
![image](https://cloud.githubusercontent.com/assets/2973219/26200279/4babb8fc-3bc5-11e7-87b2-171b4755f80e.png)
</kbd>


For reference, SSMS highlights that SQL like this:
<kbd>
![image](https://cloud.githubusercontent.com/assets/2973219/26199297/5b0d3676-3bc1-11e7-9fcd-d5c8b26b146a.png)
</kbd>

The differences between the previous behaviour and SSMS are stark and obvious.  The differences between the result of this PR and SSMS are so subtle that you won't be able to spot them.

tugberkugurlu commented:
👍 

Just curiosity: Is there any reason why we don't PR this to https://github.com/isagalaev/highlight.js?

Greg-Smulko commented:
We opened a PR to the upstream repo on 18th May, here: https://github.com/isagalaev/highlight.js/pull/1524. The latest merge to master was on 14th June. We just don't want to wait any longer.

Greg-Smulko commented:
**Info about sample usage**
We used this library for [readyroll vsts extension](https://github.com/red-gate/readyroll-vsts-extension).
In [this PR](https://github.com/red-gate/readyroll-vsts-extension/pull/16) we added highlighting using the original highlight.js library from CDN. It didn't contain the support for T-SQL, so we forked the original repo into this one and added T-SQL highlighting ourselves. We then created a [npm package](https://www.npmjs.com/package/highlight.js-tsql) and used it in [this PR](https://github.com/red-gate/readyroll-vsts-extension/pull/20).
